drivers/hmis/spec/system/hmis/form_builder_spec.rb

```diff
@@ -0,0 +1,257 @@
+require 'rails_helper'
+require_relative '../../requests/hmis/login_and_permissions'
+require_relative '../../support/hmis_base_setup'
+
+# Custom matcher for asserting order of text in Capybara page body, helpful for testing reordering of form items.
+# https://stackoverflow.com/questions/64002940/is-it-possible-to-find-out-if-an-element-comes-before-the-second-element-in-capy
+RSpec::Matchers.define :appear_before do |later_content|
+  match do |earlier_content|
+    page.body.index(earlier_content) < page.body.index(later_content)
+  end
+end
+
+RSpec.feature 'HMIS Form Builder', type: :system do
+  include_context 'hmis base setup'
+
+  let!(:ds1) { create(:hmis_data_source, hmis: 'localhost') }
+  let!(:access_control) { create_access_control(hmis_user, p1) }
+  let!(:published) { create :custom_assessment_with_custom_fields_and_rules, title: 'System Test 1', identifier: 'system_test_1', data_source: ds1 }
+  let!(:draft) { create :custom_assessment_with_custom_fields_and_rules, title: 'System Test 2', identifier: 'system_test_2', status: 'draft', data_source: ds1 }
+  let!(:advanced_draft) { create :custom_assessment_with_field_rules_and_autofill, identifier: 'advanced_form', status: 'draft', data_source: ds1 }
+
+  context 'Form builder happy path' do
+    before(:each) do
+      sign_in(hmis_user)
+      disable_transitions
+      visit '/admin/forms'
+    end
+
+    it 'creates a new form with a draft' do
+      visit '/admin/forms'
+      click_button 'New Form'
+
+      assert_text 'Form Type'
+      mui_select 'Custom assessment', from: 'Form Type'
+      fill_in 'Form Title', with: 'System Test 3'
+      fill_in 'Form Identifier', with: 'system_test_3'
+      click_button 'Save'
+
+      assert_text 'SELECTED FORM System Test 3'
+      assert_text 'Edit Draft'
+    end
+
+    it 'creates a new draft' do
+      visit '/admin/forms/system_test_1'
+      click_button 'New Draft'
+
+      assert_text 'EDITING DRAFT System Test 1'
+    end
+
+    context 'with an existing form draft' do
+      before(:each) do
+        visit "/admin/forms/system_test_2/#{draft.id}/edit"
+      end
+
+      it 'edits a form item' do
+        find("button[aria-label='section_1 item actions']").click
+        find('li', text: 'Edit', class: 'MuiMenuItem-root').trigger('click')
```

Comment:
Maybe it's confusing to add the  class here-- This was to disambiguate; capybara returns an ambiguity error without it. I was not sure what capybara thought was ambiguous, since I couldn't see any other obvious `li` with text "Edit" on the page